File: Leetcode100/leetcode100_python/CountingBits.py
from typing import List

class Solution:
    def countBits(self, num: int) -> List[int]:
        bits = [0] * (num + 1)
        for i in range(num + 1):
            count = 0
            n = i
            while n:
                count += 1
                n &= n - 1
            bits[i] = count
        return bits

if __name__ == "__main__":
    num = int(input())
    solution = Solution()
    print(solution.countBits(num))

File: Leetcode100/leetcode100/src/main/java/com/jiang/leetcode100/counting_bits/CountingBits.java
package main.java.com.jiang.leetcode100.counting_bits;

import java.util.Scanner;

class Solution {
    public int[] countBits(int num) {
        int[] bits = new int[num + 1];
        for(int i = 0; i <= num; i++) {
            int count = 0, n = i;
            while(n > 0) {
                if((n & 1) > 0) count++;
                n >>= 1;
            }
            bits[i] = count;
        }
        return bits;
    }
}

public class CountingBits {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int num = scanner.nextInt();
        scanner.close();
        int[] result = new Solution().countBits(num);
        for(int n : result) {
            System.out.print(n + " ");
        }
    }
}